File: wrkspc/kana-core/src/parser/glyphs/punctuation.rs
use phf::{phf_map as map, Map};

pub static MARKS: Map<&'static str, &'static str> = map! {
	">" => "】",
	"<" => "【",
	"]" => "］",
	"[" => "［",
	"(" => "（",
	")" => "）",
	"{" => "｛",
	"}" => "｝",
	"..." => "…",
	"," => "、",
	"=" => "＝",
	"." => "。",
	":" => "：",
	"!" => "！",
	"?" => "？",
};

pub const SINGLE_QUOTES: [&'static str; 2] = ["「", "」"];
pub const DOUBLE_QUOTES: [&'static str; 2] = ["『", "』"];

File: src/punctuation.zig
const std = @import("std");

const KV = struct { @"0": []const u8, @"1": []const u8 };
pub const Marks = std.ComptimeStringMap([]const u8, [_]KV{
    .{ ">", "】" },
    .{ "<", "【" },
    .{ "]", "］" },
    .{ "[", "［" },
    .{ "(", "（" },
    .{ ")", "）" },
    .{ "{", "｛" },
    .{ "}", "｝" },
    .{ "...", "…" },
    .{ ",", "、" },
    .{ "=", "＝" },
    .{ ".", "。" },
    .{ ":", "：" },
    .{ "!", "！" },
    .{ "?", "？" },
    .{ "~", "〜" },
});

pub const single_quotes = [_][]const u8{ "「", "」" };
pub const double_quotes = [_][]const u8{ "『", "』" };
